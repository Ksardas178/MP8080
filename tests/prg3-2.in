/*Рассмотрите сложение двоично-десятичных
чисел. Рассмотрите действие команды DAA 
для различных значений суммируемых операндов.
Результат выполнения фиксируйте в памяти.*/

//===========================================(36,27)
//loading 1st num
	//loading first digit
	MVI A 3
	//loading second digit
	RLC RLC RLC RLC
	MVI C 6
	ADD C //A <- A+C
	//0011|0110
//Moving data
MOV B A
//loading 2nd num
	//loading first digit
	MVI A 2
	//loading second digit
	RLC RLC RLC RLC
	MVI C 7
	ADD C //A <- A+C
//now we have 36 and 27 in B and A
//0011|0110
//0010|0111
ADD B	//A <- A + B
DAA		//correction (A)
STA o200 o000 //store A
//===========================================(15,25)		
HLT
		









