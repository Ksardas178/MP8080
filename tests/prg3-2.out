
[32;1mInput is correct
[0m
[30;1mCode analysis results:
[0m______________________________
00000000: 00111110
00000001: 00000011
00000010: 00000111
00000011: 00000111
00000100: 00000111
00000101: 00000111
00000110: 00001110
00000111: 00000110
00001000: 10000001
00001001: 01000111
00001010: 00111110
00001011: 00000010
00001100: 00000111
00001101: 00000111
00001110: 00000111
00001111: 00000111
00010000: 00001110
00010001: 00000111
00010010: 10000001
00010011: 10000000
00010100: 00100111
00010101: 00110010
00010110: 10000000
00010111: 00000000
00011000: 01110110
